File: tests/benchmark/fold/ocaml/fold.ml
(* fold a list of N integers *)

let plusMod x y = (x + y) mod 268435456

let rec run n acc lst =
  if n == 0 then
    List.fold_left plusMod acc lst
  else
    run (n - 1) (List.fold_left plusMod acc lst) lst

let rec gen n acc =
  if n == 0 then
    acc
  else
    gen (n - 1) (n :: acc)

;;

let k = 1000
and n = 100000
in

print_int (run k 0 (gen n []));
print_newline ();

File: tests/benchmark/fold/haskell/fold.hs
-- fold a list of N integers

import Data.List

plusMod :: Int -> Int -> Int
plusMod x y = (x + y) `mod` (2 ^ 28)

run :: Int -> Int -> [Int] -> Int
run 0 acc lst = foldl' plusMod acc lst
run n acc lst = run (n - 1) (foldl' plusMod acc lst) lst

main :: IO ()
main = putStrLn (show (run k 0 [1 .. n]))
  where
    k = 1000
    n = 100000
